### PR TITLE
fix: allow empty responses

### LINKED ISSUE #4833: Allow empty responses
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

Under the [OpenAPI 3.0 specifications](https://swagger.io/docs/specification/describing-responses/), "Empty Response Body" explains that responses such as 204 No Content can omit the `content` and just have a description. This should be an allowed feature.

### Motivation

This is useful as it covers the full specification.

### Example

I have tried to do this...
```js
{
    schema: {
        ...
        response: {
            204: {
                description: "Session deleted"
            }
        }
    }
}
```
...but it currently errors
```
FastifyError [Error]: Failed building the serialization schema for DELETE: /users/sessions/@me, due to error schema is invalid: data/properties/description must be object,boolean
```
I have to include a `type: "object"` and empty `properties: {}` for it to build. Having only a description should be supported.

### PR DESCRIPTION
**Pull request by giuliowaitforitdavide**
Closes #4833 

#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

#### Other Information

I decided to include the condition `is204` and assume it's an empty object (if we are not able to retrieve the schema) based on [Mozilla Developer docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204)

### CODE REVIEW

File: lib/schemas.js [hunk 1]
```diff
@@ -129,8 +129,14 @@ function generateFluentSchema (schema) {
   }
 }
 
-function getSchemaAnyway (schema, jsonShorthand) {
+function getSchemaAnyway (schema, jsonShorthand, is204) {
   if (!jsonShorthand || schema.$ref || schema.oneOf || schema.allOf || schema.anyOf || schema.$merge || schema.$patch) return schema
+  if (!schema.type && !schema.properties && is204) {
+    return {
+      type: 'object',
```
Comment (Eomm) [inline lib/schemas.js:136]: This identifies an empty object `{}` as response - not an empty string tho

File: lib/schemas.js [hunk 2]
```diff
@@ -103,7 +103,7 @@ function normalizeSchema (routeSchemas, serverOptions) {
       }
 
       if (!hasContentMultipleContentTypes) {
-        routeSchemas.response[code] = getSchemaAnyway(routeSchemas.response[code], serverOptions.jsonShorthand)
+        routeSchemas.response[code] = getSchemaAnyway(routeSchemas.response[code], serverOptions.jsonShorthand, code === '204')
```
Comment (Eomm) [inline lib/schemas.js:106]: I would follow a different path here.

I would ignore the schema for this point:
https://github.com/fastify/fastify/blob/fd62068892b7312fcb8552834fbf60a035442ad3/lib/schemas.js#L154

I think it worth to say "if the standard does not expect a payload and the object does not contain the properties, just ignore it"

Comment (giuliowaitforitdavide) [reply lib/schemas.js:106]: Maybe you meant to mention the `getSchemaAnyway` function?

**Comment by Uzlopak** [comment]
Why closed?

**Comment by giuliowaitforitdavide** [comment]
Are we working on #4865 ? The PRs are for the same issue with the same code, I thought this was not needed

**Comment by Uzlopak** [comment]
Imho nothing is decided. We should determine first how to solve it.

**Pull request closed without merging**